src/index.ts

```diff
@@ -1,39 +1,57 @@
 import debug = require('debug')
-import flatten = require('array-flatten')
 
 const log = debug('compose-middleware')
 
 export type Next<T = void> = (err?: Error | null) => T
 export type RequestHandler <T, U, V = void> = (req: T, res: U, next: Next<V>) => V
-export type ErrorHandler <T, U, V = void> = (err: Error, req: T, res: U, next: Next<V>) => V
+export type ErrorHandler <T, U, V = void> = (err: Error | null, req: T, res: U, next: Next<V>) => V
 export type Middleware <T, U, V = void> = RequestHandler<T, U, V> | ErrorHandler<T, U, V>
 
-export type Handler <T, U, V = void> = Middleware<T, U, V> | flatten.NestedArray<Middleware<T, U, V>>
+/**
+ * Compose a variadic number of middlewares into a single middleware.
+ */
+export function compose <T, U, V = void> (
```

Comment:
I can. This is the curre API name changes:

```
errors -> errors
compose -> successes
            -> compose (new)
```
However I can do:
```
errors -> errors
compose -> compose
          -> pipe (new)
```
I like `pipe` because makes me think I applying middlewares left to right. However if we have this final API, `compose` makes me think that makes the same as `pipe` but right to left. For this reason, I prefered having an analog of `errors` that works with request/success handlers.